internal/pkg/docker/dockercompose/svc.go

```diff
@@ -96,10 +101,68 @@ type exposedPort struct {
 }
 
 // findExposedPort attempts to detect a singular exposed port in the given service and determines if it is publicly exposed.
-func findExposedPort(service *compose.ServiceConfig) (*exposedPort, error) {
-	// TODO: Port handling & exposed port detection, to be implemented in Milestone 3
+func findExposedPort(service *compose.ServiceConfig, workingDir string) (*exposedPort, error) {
+	if len(service.Ports) == 1 {
+		binding := service.Ports[0]
+		port := uint16(binding.Target)
+
+		if binding.Published != "" && strconv.Itoa(int(port)) != binding.Published {
+			return nil, fmt.Errorf("cannot publish the container port %v under a different public port %v in Copilot", port, binding.Published)
+		}
+
+		return &exposedPort{
+			port:   port,
+			public: true,
```

Comment:
Just wondering if we need to check if `binding.Published` isn't empty before knowing this is public...?

Reply:
So this is actually a weird edge case in the spec! Here's what [the spec](https://docs.docker.com/compose/compose-file/#ports) says about it... so I think it should be fair to make it the same as the container port here, but I do need to add this check! Good catch 👍 

> Either specify both ports (HOST:CONTAINER), or just the container port. In the latter case, the Compose implementation SHOULD automatically allocate any unassigned host port.